### ISSUE TITLE
New Option in `no-missing-import`: Ability to allow Bun modules

### ISSUE THREAD

**Issue by regseb** [weiran-zsd/eslint-plugin-node#494]
### Rule details

```javascript
import test from "bun:test";
```

The rule [`no-missing-import`](https://github.com/eslint-community/eslint-plugin-n/blob/v17.23.1/docs/rules/no-missing-import.md) reports that it can't resolve 'bun:test'.

If I add this option:

```javascript
"n/no-missing-import": ["error", {
    allowModules: ["bun:test"],
}],
```

I get the error:
> Error: Key "rules": Key "n/no-missing-import":
>         Value "bun:test" should match pattern "^(?:virtual:)?(?:@[a-zA-Z0-9_\-.]+/)?[a-zA-Z0-9_\-.]+$".

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

- `index.js`

  ```javascript  
  import test from "bun:test";
  ```

- `package.json`

  ```json
  {
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
      "eslint": "9.39.2",
      "eslint-plugin-n": "17.23.1"
    }
  }
  ```

- `eslint.config.js`

  ```javascript
  import n from "eslint-plugin-n";

  export default [{
      plugins: { n },

      rules: {
          "n/no-missing-import": ["error", {
              allowModules: ["bun:test"],
          }],
      },
  }];
  ```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

I disabled this rule to avoid false positives. But I would like to enable it to always check Node modules imported into files executed in the Bun runtime.

**Comment by Milanz505** [comment]
Hi! I'm a Computer Science student and I'm currently studying Compilers and Static Analysis. I'd like to investigate this false positive in the no-missing-import rule. Could you please assign this to me?

**Comment by aladdin-add** [comment]
This seems like an interesting proposal. Currently, this plugin only targets code running on Node.js and does not include Bun or other JavaScript runtimes. Should we expand its scope of use?
